### ISSUE TITLE
Non-default channel does not work

### ISSUE THREAD

**Issue by solicomo** [wbond/package_control#935]
I create a [private channel](https://raw.githubusercontent.com/solicomo/sublime-package-control-channel/master/channel.json) according to the [doc](https://packagecontrol.io/docs/channels_and_repositories), and imitating the [default channel](https://github.com/wbond/package_control_channel).

Then I added it to Package Control:

```json
{
	"channels":
	[
		"https://packagecontrol.io/channel_v3.json",
		"https://raw.githubusercontent.com/solicomo/sublime-package-control-channel/master/channel.json"
	],

	//...
}
```

But when I try to install package, the packages in my private channel do not show up.

**Comment by wbond** [comment]
Please submit a debug log.

**Comment by solicomo** [comment]
Here is the debug log:

```
Package Control: Fetching list of available packages
  Platform: windows-x64
  Sublime Text Version: 3083
  Package Control Version: 3.0.1
Package Control: Download Debug
  URL: https://packagecontrol.io/channel_v3.json
  Resolved IPv6: 2600:3c02::f03c:91ff:fedf:6883
  Resolved IP: 50.116.33.29
  Timeout: 30
Package Control: WinINet Debug Proxy
  proxy: 
  proxy bypass: 
  proxy username: 
  proxy password: 
Package Control: WinINet HTTPS Debug General
  Server SSL Certificate:
    subject: US, Massachusetts, Private Organization, 471714639, US, Massachusetts, Newbury, Codex Non Sufficit LC, codexns.io
    issuer: US, GeoTrust Inc., GeoTrust EV SSL CA - G4
    common name: codexns.io
    issue date: Wed, 17 Dec 2014 08:00:00 GMT
    expire date: Fri, 30 Oct 2015 07:59:59 GMT
Package Control: WinINet HTTPS Debug Write
  GET /channel_v3.json HTTP/1.1
  Accept-Encoding: bzip2,gzip,deflate
  If-Modified-Since: Wed, 22 Apr 2015 03:00:05 GMT
  If-None-Match: "55370eb5-32e6e"
  User-Agent: Package Control v3.0.1
  Host: packagecontrol.io
  Connection: Keep-Alive
  Cache-Control: no-cache
Package Control: WinINet HTTPS Debug Read
  HTTP/1.1 200 OK
  Server: nginx
  Date: Wed, 22 Apr 2015 03:23:34 GMT
  Content-Type: application/json
  Content-Length: 206082
  Last-Modified: Wed, 22 Apr 2015 03:20:03 GMT
  Connection: keep-alive
  ETag: "55371363-32502"
  Content-Encoding: bzip2
  Vary: Accept-Encoding
  Strict-Transport-Security: max-age=15552000
  Accept-Ranges: bytes
Package Control: Caching https://packagecontrol.io/channel_v3.json in D:\Program Files\SublimeText\Data\Packages\User\Package Control.cache\01524fae79697630d0454ba3fabd9414
Package Control: Download Debug
  URL: https://raw.githubusercontent.com/solicomo/sublime-package-control-channel/master/channel.json
  Resolved IP: 199.27.75.133
  Timeout: 30
Package Control: WinINet Debug Proxy
  proxy: 
  proxy bypass: 
  proxy username: 
  proxy password: 
Package Control: WinINet HTTPS Debug General
  Server SSL Certificate:
    subject: US, San Francisco, California, "Fastly, Inc.", www.github.com
    issuer: US, DigiCert Inc, www.digicert.com, DigiCert SHA2 High Assurance Server CA
    common name: www.github.com
    issue date: Mon, 23 Feb 2015 08:00:00 GMT
    expire date: Wed, 02 Mar 2016 20:00:00 GMT
Package Control: WinINet HTTPS Debug Write
  GET /solicomo/sublime-package-control-channel/master/channel.json HTTP/1.1
  Accept-Encoding: bzip2,gzip,deflate
  If-None-Match: "2b3f36b5af4176b96efe389e654500e2bf838daf"
  User-Agent: Package Control v3.0.1
  Host: raw.githubusercontent.com
  Connection: Keep-Alive
  Cache-Control: no-cache
Package Control: WinINet HTTPS Debug Read
  HTTP/1.1 304 Not Modified
  Date: Wed, 22 Apr 2015 03:23:38 GMT
  Via: 1.1 varnish
  Last-Modified: Wed, 22 Apr 2015 03:23:39 GMT
  Cache-Control: max-age=300
  ETag: "2b3f36b5af4176b96efe389e654500e2bf838daf"
  Connection: keep-alive
  X-Served-By: cache-den6025-DEN
  X-Cache: MISS
  X-Cache-Hits: 0
  Vary: Authorization,Accept-Encoding
  Expires: Wed, 22 Apr 2015 03:28:39 GMT
  Source-Age: 0
Package Control: Using cached content for https://raw.githubusercontent.com/solicomo/sublime-package-control-channel/master/channel.json from D:\Program Files\SublimeText\Data\Packages\User\Package Control.cache\7f3ce864b77e5cc8f511fcfb016960e3
Exception in thread Thread-8:
Traceback (most recent call last):
  File "./threading.py", line 901, in _bootstrap_inner
  File "package_control.automatic_upgrader in D:\Program Files\SublimeText\Data\Installed Packages\Package Control.sublime-package", line 111, in run
  File "package_control.automatic_upgrader in D:\Program Files\SublimeText\Data\Installed Packages\Package Control.sublime-package", line 165, in install_missing
  File "package_control.package_manager in D:\Program Files\SublimeText\Data\Installed Packages\Package Control.sublime-package", line 343, in list_available_packages
  File "package_control.package_manager in D:\Program Files\SublimeText\Data\Installed Packages\Package Control.sublime-package", line 306, in list_repositories
  File "package_control.providers.channel_provider in D:\Program Files\SublimeText\Data\Installed Packages\Package Control.sublime-package", line 168, in get_renamed_packages
KeyError: 'packages_cache'

Package Control: WinINet HTTPS Debug General
  Closing connection to packagecontrol.io on port 443 after 1 request
Package Control: WinINet HTTPS Debug General
  Closing connection to raw.githubusercontent.com on port 443 after 1 request
WINDOW COMMAND ENABLED True
WINDOW COMMAND ENABLED True
WINDOW COMMAND ENABLED True
WINDOW COMMAND ENABLED True
WINDOW COMMAND ENABLED True
WINDOW COMMAND ENABLED False
Package Control: Fetching list of available packages
  Platform: windows-x64
  Sublime Text Version: 3083
  Package Control Version: 3.0.1
```

**Comment by wbond** [comment]
I've fixed the issue in 2066136adfd6d7d00a958cb58b7a13aaa1d15a72. Alternatively, you can create a `packages_cache` entry in your channel that is just `{}`.
